examples/bri-3/src/bri/communication/agents/messaging.agent.ts

```diff
@@ -0,0 +1,34 @@
+import { Inject, Injectable, OnApplicationBootstrap } from '@nestjs/common';
+import { CommandBus } from '@nestjs/cqrs';
+import { LoggingService } from '../../../shared/logging/logging.service';
+import { ProcessInboundMessageCommand } from '../capabilities/processInboundMessage/processInboundMessage.command';
+import { IMessagingClient } from '../messagingClients/messagingClient.interface';
+
+@Injectable()
+export class MessagingAgent implements OnApplicationBootstrap {
+  constructor(
+    @Inject('IMessagingClient')
```

Comment:
this looks like a new syntax, why do we need @Inject decorator?

Reply:
It is necessary because the InjectionToken 'IMessagingClient' is defined as a string in the CommunicationModule providers definition, which is done because we want  to provide different clients based on config later (i.e. NATS, RabbitMQ ...).